src/Access/User.cpp

```diff
@@ -11,7 +11,10 @@ bool User::equal(const IAccessEntity & other) const
     const auto & other_user = typeid_cast<const User &>(other);
     return (authentication == other_user.authentication) && (allowed_client_hosts == other_user.allowed_client_hosts)
         && (access == other_user.access) && (granted_roles == other_user.granted_roles) && (default_roles == other_user.default_roles)
-        && (settings == other_user.settings) && (grantees == other_user.grantees);
+        && (settings == other_user.settings) && (grantees == other_user.grantees)
+        /// Daisy : start
+        &&(database == other_user.database);
```

Comment:
Can we avoid doing user to database mapping at low level ? I think probably we will run into issue in the future.

Reply:
1.Modifying the code at low level  is to dynamically modify the mapped database according to the configuration
2.There is another solution here, which is to put the mapping information in config.xml and re-read the mapping every time, but I think it would be more appropriate to put it in user.xml
3.If you only mean to do code-level mapping temporarily, this modification can be removed, and only human flesh matches in the code to solve this layer of mapping